plugins/inputs/mavlink/mavlink.go

```diff
@@ -0,0 +1,196 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mavlink
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/bluenviron/gomavlib/v3"
+	"github.com/bluenviron/gomavlib/v3/pkg/dialects/ardupilotmega"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Mavlink struct {
+	URL                    string   `toml:"url"`
+	SystemID               uint8    `toml:"system_id"`
+	FilterPattern          []string `toml:"filter"`
+	StreamRequestEnable    bool     `toml:"stream_request_enable"`
+	StreamRequestFrequency int      `toml:"stream_request_frequency"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	filter         filter.Filter
+	connection     *gomavlib.Node
+	endpointConfig []gomavlib.EndpointConf
+	cancel         context.CancelFunc
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Mavlink) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Mavlink) Init() error {
+	// Parse out the Mavlink endpoint.
+	// Try to parse the URL
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return fmt.Errorf("invalid url: %w", err)
+	}
```

Comment:
You should set defaults here instead of doing it in `init`. This ensures that defaults are respected even-though users explicitly set the options to empty values...
```suggestion
	// Set defaults
	if s.URL == "" {
		s.URL = "udp://:14540"
	}
	
	if s.StreamRequestEnable && s.StreamRequestFrequency == 0 {
		s.StreamRequestFrequency = 4
	}
	
	// Check the input parameters
	u, err := url.Parse(s.URL)
	if err != nil {
		return fmt.Errorf("invalid url: %w", err)
	}
```

Reply:
Yes, makes sense, I've added this so that any of the defaults are set inside Init (the URL and stream rates)